os/fs/driver/mtd/mtd_partition.c

```diff
@@ -572,27 +574,21 @@ static ssize_t part_procfs_read(FAR struct file *filep, FAR char *buffer, size_t
 		/* Copy data from the next known partition */
 
 #ifdef CONFIG_MTD_PARTITION_NAMES
-		if (attr->nextpart->name == NULL) {
-			strncpy(partname, "(noname)  ", MTD_PARTNAME_LEN);
-		} else {
-			ptr = attr->nextpart->name;
-			for (x = 0; x < sizeof(partname) - 1; x++) {
-				/* Test for end of partition name */
-
-				if (*ptr == ',' || *ptr == '\0') {
-					/* Perform space fill for alignment */
-
-					partname[x] = ' ';
-				} else {
-					/* Copy next byte of partition name */
-
-					partname[x] = *ptr++;
-				}
+		ptr = attr->nextpart->name;
+		for (x = 0; x < sizeof(partname) - 1; x++) {
```

Comment:
How about :

uint16_t x_max = sizeof(partname) - 1;
for (x = 0; x < x_max; x++) {

By this a variable memory will increase but This will be more optimize as it didn't call " sizeof " fun over n over again.

Reply:
Okay i modified to MTD_PARTNAME_LEN